MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm

```diff
@@ -100,12 +100,18 @@
 	}
 
 	@autoreleasepool {
+		NSArray* mtlDevs = [MTLCopyAllDevices() autorelease];
+		if (mtlDevs.count == 0) {
+			errMsg = "Could not retrieve MTLDevice to compile shader.";
+			return false;
+		}
+
 		MTLCompileOptions* mtlCompileOptions  = [[MTLCompileOptions new] autorelease];
 		mtlCompileOptions.languageVersion = mslVerEnum;
 		NSError* err = nil;
-		id<MTLLibrary> mtlLib = [[MTLCreateSystemDefaultDevice() newLibraryWithSource: @(mslSourceCode.c_str())
-																			  options: mtlCompileOptions
-																				error: &err] autorelease];
+		id<MTLLibrary> mtlLib = [[mtlDevs[0] newLibraryWithSource: @(mslSourceCode.c_str())
+														  options: mtlCompileOptions
+															error: &err] autorelease];
```

Comment:
This change does not match the description in the commit message, which confused me.

Reply:
Oh geez. I got the commit message description backwards. Fixed now.

Reply:
Oh, geez. I got the commit message description wrong. Fixed now.